### PR TITLE
Make Pod SecurityContext Mutation more accurate

### PR DESCRIPTION
**Pull request by cbarbian-sap**
Currently (if pre-conditions are fulfilled),  Karydia will override an existing pod security context, and just set runAsUser and runAsGroup. Other previously existing attributes (such as fsGroup) will get lost. Which probably should not be the case, as Karydia claims to only own runAsUser and runAsGroup.

This PR tries to fix this behaviour, so that in an existing pod security context, only runAsUser and runAsGroup will be patched by Karydia.

A similar behaviour was observed in the logic where Karydia patches existing container security contexts. This is also covered by this PR.

### CODE REVIEW

**Comment by Neumann-Nils** [comment]
Looks good to me and worked while testing it manually. Needs unit tests (https://github.com/karydia/karydia/blob/master/pkg/admission/karydia/admission_test.go) and integration tests (https://github.com/karydia/karydia/blob/master/tests/e2e/admission_security_context_test.go).

**Comment by cbarbian-sap** [comment]
Unit test and integration tests have been added

File: pkg/admission/karydia/admission_pod.go [hunk 1]
```diff
@@ -128,18 +128,36 @@ func mutatePodSecurityContext(pod corev1.Pod, setting Setting, patches Patches)
 				},
 			})
 			annotatePod(pod, &patches, "karydia.gardener.cloud/podSecurityContext.internal", setting.src+"/"+setting.value)
+		} else if secCtx.RunAsUser == nil && secCtx.RunAsGroup == nil {
+			patches.operations = append(patches.operations, patchOperation{
+				Op:    "add",
+				Path:  "/spec/securityContext/runAsUser",
+				Value: uid,
```
Comment (Neumann-Nils) [inline pkg/admission/karydia/admission_pod.go:135]: For consistency, pass by pointer: Reference variables with `&uid` as in the coding before.

File: pkg/admission/karydia/admission_pod.go [hunk 2]
```diff
@@ -128,18 +128,36 @@ func mutatePodSecurityContext(pod corev1.Pod, setting Setting, patches Patches)
...
 					Path: "/spec/containers/" + strconv.Itoa(i) + "/securityContext",
 					Value: corev1.SecurityContext{
 						AllowPrivilegeEscalation: &privilegeEscalation,
 					},
 				})
+			} else if secCtxContainers.AllowPrivilegeEscalation == nil {
+				patches.operations = append(patches.operations, patchOperation{
+					Op:    "add",
+					Path:  "/spec/containers/" + strconv.Itoa(i) + "/securityContext/allowPrivilegeEscalation",
+					Value: privilegeEscalation,
```
Comment (Neumann-Nils) [inline pkg/admission/karydia/admission_pod.go:159]: For consistency, pass by pointer: Reference variable with `&privilegeEscalation` as in the coding before.

File: pkg/admission/karydia/admission_pod.go (hunk 1 above)
Comment (Neumann-Nils) [inline pkg/admission/karydia/admission_pod.go:135]: For consistency, pass by pointer: Reference variables with `&uid` as in the coding before.

File: pkg/admission/karydia/admission_pod.go [hunk 3]
```diff
@@ -128,18 +128,36 @@ func mutatePodSecurityContext(pod corev1.Pod, setting Setting, patches Patches)
...
+		} else if secCtx.RunAsUser == nil && secCtx.RunAsGroup == nil {
+			patches.operations = append(patches.operations, patchOperation{
+				Op:    "add",
+				Path:  "/spec/securityContext/runAsUser",
+				Value: uid,
+			})
+			patches.operations = append(patches.operations, patchOperation{
+				Op:    "add",
+				Path:  "/spec/securityContext/runAsGroup",
+				Value: gid,
```
Comment (Neumann-Nils) [inline pkg/admission/karydia/admission_pod.go:140]: For consistency, pass by pointer: Reference variables with `&gid` as in the coding before.

File: pkg/admission/karydia/admission_pod.go (hunk 2 above)
Comment (Neumann-Nils) [reply pkg/admission/karydia/admission_pod.go:159]: Pass by value should be the better option.

File: pkg/admission/karydia/admission_pod.go (hunk 1 above)
Comment (Neumann-Nils) [reply pkg/admission/karydia/admission_pod.go:135]: Pass by value should be the better option.

File: pkg/admission/karydia/admission_pod.go (hunk 3 above)
Comment (Neumann-Nils) [reply pkg/admission/karydia/admission_pod.go:140]: Pass by value should be the better option.

**Pull request merged**